oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/expression/BooleanMatch.java

```diff
@@ -23,7 +23,7 @@
 @BooleanValueFilterMatcher
 public class BooleanMatch {
     public boolean match(Boolean left, Boolean right) {
-        return left == right;
+        return left.equals(right);
```

Comment:
When `left == null`, NPE is thrown

```suggestion
        return Objects.equals(left, right);
```